lib/AllocateDescriptorsPass.cpp

```diff
@@ -864,3 +899,106 @@ bool AllocateDescriptorsPass::AllocateLocalKernelArgSpecIds(Module &M) {
 
   return Changed;
 }
+
+bool AllocateDescriptorsPass::CallTreeContainsGlobalBarrier(Function *F) {
```

Comment:
Is it worth saying somewhere that because we're assuming OpenCL 1.2, atomics are always Relaxed.  So they don't synchronize.  So the presence of an atomic op does not require the addition of a Coherent decoration.